### PR TITLE
dom/met signature: imputers

### PR DESCRIPTION
**Pull request by Shoeboxam**

### CODE REVIEW

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #696** <a href="https://app.graphite.dev/github/pr/opendp/opendp/696" target="_blank">Graphite</a> 
    * **PR #645** <a href="https://app.graphite.dev/github/pr/opendp/opendp/645" target="_blank">Graphite</a> 
      * **PR #691** <a href="https://app.graphite.dev/github/pr/opendp/opendp/691" target="_blank">Graphite</a> 
      * **PR #604** <a href="https://app.graphite.dev/github/pr/opendp/opendp/604" target="_blank">Graphite</a> 
        * **PR #706** <a href="https://app.graphite.dev/github/pr/opendp/opendp/706" target="_blank">Graphite</a> 
        * **PR #699** <a href="https://app.graphite.dev/github/pr/opendp/opendp/699" target="_blank">Graphite</a> 
        * **PR #687** <a href="https://app.graphite.dev/github/pr/opendp/opendp/687" target="_blank">Graphite</a> 
          * **PR #688** <a href="https://app.graphite.dev/github/pr/opendp/opendp/688" target="_blank">Graphite</a> 
            * **PR #689** <a href="https://app.graphite.dev/github/pr/opendp/opendp/689" target="_blank">Graphite</a> 
              * **PR #749** <a href="https://app.graphite.dev/github/pr/opendp/opendp/749" target="_blank">Graphite</a> 
              * **PR #712** <a href="https://app.graphite.dev/github/pr/opendp/opendp/712" target="_blank">Graphite</a> 
                * **PR #713** <a href="https://app.graphite.dev/github/pr/opendp/opendp/713" target="_blank">Graphite</a> 
                  * **PR #736** <a href="https://app.graphite.dev/github/pr/opendp/opendp/736" target="_blank">Graphite</a> 
                    * **PR #761** <a href="https://app.graphite.dev/github/pr/opendp/opendp/761" target="_blank">Graphite</a> 
                      * **PR #798** <a href="https://app.graphite.dev/github/pr/opendp/opendp/798" target="_blank">Graphite</a> 
                        * **PR #799** <a href="https://app.graphite.dev/github/pr/opendp/opendp/799" target="_blank">Graphite</a> 
                          * **PR #800** <a href="https://app.graphite.dev/github/pr/opendp/opendp/800" target="_blank">Graphite</a> 
                            * **PR #801** <a href="https://app.graphite.dev/github/pr/opendp/opendp/801" target="_blank">Graphite</a> 
                              * **PR #802** <a href="https://app.graphite.dev/github/pr/opendp/opendp/802" target="_blank">Graphite</a> 
                                * **PR #803** <a href="https://app.graphite.dev/github/pr/opendp/opendp/803" target="_blank">Graphite</a> 
                                  * **PR #804** <a href="https://app.graphite.dev/github/pr/opendp/opendp/804" target="_blank">Graphite</a> 
                                    * **PR #806** <a href="https://app.graphite.dev/github/pr/opendp/opendp/806" target="_blank">Graphite</a> 
                                      * **PR #807** <a href="https://app.graphite.dev/github/pr/opendp/opendp/807" target="_blank">Graphite</a> 
                                        * **PR #808** <a href="https://app.graphite.dev/github/pr/opendp/opendp/808" target="_blank">Graphite</a> 
                                          * **PR #810** <a href="https://app.graphite.dev/github/pr/opendp/opendp/810" target="_blank">Graphite</a> 
                                            * **PR #813** <a href="https://app.graphite.dev/github/pr/opendp/opendp/813" target="_blank">Graphite</a>  👈
                                              * **PR #814** <a href="https://app.graphite.dev/github/pr/opendp/opendp/814" target="_blank">Graphite</a> 
                                                * **PR #815** <a href="https://app.graphite.dev/github/pr/opendp/opendp/815" target="_blank">Graphite</a> 
                      * **PR #796** <a href="https://app.graphite.dev/github/pr/opendp/opendp/796" target="_blank">Graphite</a> 
              * **PR #690** <a href="https://app.graphite.dev/github/pr/opendp/opendp/690" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/813?utm_source=stack-comment).

**Comment by Shoeboxam** [comment]
Current dependencies on/for this PR:
* main
  * **PR #696** <a href="https://app.graphite.dev/github/pr/opendp/opendp/696" target="_blank">Graphite</a> 
    * **PR #645** <a href="https://app.graphite.dev/github/pr/opendp/opendp/645" target="_blank">Graphite</a> 
      * **PR #691** <a href="https://app.graphite.dev/github/pr/opendp/opendp/691" target="_blank">Graphite</a> 
      * **PR #604** <a href="https://app.graphite.dev/github/pr/opendp/opendp/604" target="_blank">Graphite</a> 
        * **PR #706** <a href="https://app.graphite.dev/github/pr/opendp/opendp/706" target="_blank">Graphite</a> 
        * **PR #699** <a href="https://app.graphite.dev/github/pr/opendp/opendp/699" target="_blank">Graphite</a> 
        * **PR #687** <a href="https://app.graphite.dev/github/pr/opendp/opendp/687" target="_blank">Graphite</a> 
          * **PR #688** <a href="https://app.graphite.dev/github/pr/opendp/opendp/688" target="_blank">Graphite</a> 
            * **PR #689** <a href="https://app.graphite.dev/github/pr/opendp/opendp/689" target="_blank">Graphite</a> 
              * **PR #749** <a href="https://app.graphite.dev/github/pr/opendp/opendp/749" target="_blank">Graphite</a> 
              * **PR #712** <a href="https://app.graphite.dev/github/pr/opendp/opendp/712" target="_blank">Graphite</a> 
                * **PR #713** <a href="https://app.graphite.dev/github/pr/opendp/opendp/713" target="_blank">Graphite</a> 
                  * **PR #736** <a href="https://app.graphite.dev/github/pr/opendp/opendp/736" target="_blank">Graphite</a> 
                    * **PR #761** <a href="https://app.graphite.dev/github/pr/opendp/opendp/761" target="_blank">Graphite</a> 
                      * **PR #798** <a href="https://app.graphite.dev/github/pr/opendp/opendp/798" target="_blank">Graphite</a> 
                        * **PR #799** <a href="https://app.graphite.dev/github/pr/opendp/opendp/799" target="_blank">Graphite</a> 
                          * **PR #800** <a href="https://app.graphite.dev/github/pr/opendp/opendp/800" target="_blank">Graphite</a> 
                            * **PR #801** <a href="https://app.graphite.dev/github/pr/opendp/opendp/801" target="_blank">Graphite</a> 
                              * **PR #802** <a href="https://app.graphite.dev/github/pr/opendp/opendp/802" target="_blank">Graphite</a> 
                                * **PR #803** <a href="https://app.graphite.dev/github/pr/opendp/opendp/803" target="_blank">Graphite</a> 
                                  * **PR #804** <a href="https://app.graphite.dev/github/pr/opendp/opendp/804" target="_blank">Graphite</a> 
                                    * **PR #806** <a href="https://app.graphite.dev/github/pr/opendp/opendp/806" target="_blank">Graphite</a> 
                                      * **PR #807** <a href="https://app.graphite.dev/github/pr/opendp/opendp/807" target="_blank">Graphite</a> 
                                        * **PR #808** <a href="https://app.graphite.dev/github/pr/opendp/opendp/808" target="_blank">Graphite</a> 
                                          * **PR #810** <a href="https://app.graphite.dev/github/pr/opendp/opendp/810" target="_blank">Graphite</a> 
                                            * **PR #813** <a href="https://app.graphite.dev/github/pr/opendp/opendp/813" target="_blank">Graphite</a>  👈
                                              * **PR #814** <a href="https://app.graphite.dev/github/pr/opendp/opendp/814" target="_blank">Graphite</a> 
                                                * **PR #815** <a href="https://app.graphite.dev/github/pr/opendp/opendp/815" target="_blank">Graphite</a> 
                                                  * **PR #816** <a href="https://app.graphite.dev/github/pr/opendp/opendp/816" target="_blank">Graphite</a> 
                      * **PR #796** <a href="https://app.graphite.dev/github/pr/opendp/opendp/796" target="_blank">Graphite</a> 
              * **PR #690** <a href="https://app.graphite.dev/github/pr/opendp/opendp/690" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/opendp/opendp/813?utm_source=stack-comment).

File: rust/src/transformations/impute/mod.rs [hunk 1]
```diff
@@ -3,34 +3,46 @@ mod ffi;
...
+
+#[bootstrap(
+    features("contrib"), 
+    generics(M(suppress), TA(suppress)),
+    derived_types(TA = "$get_atom(get_type(input_domain))")
+)]
 /// Make a Transformation that replaces NaN values in `Vec<TA>` with uniformly distributed floats within `bounds`.
 ///
 /// # Arguments
+/// * `input_domain` - Domain of the input.
```
Comment (andrewvyrros) [inline rust/src/transformations/impute/mod.rs:23]: Do you want a table like other constructors above here?

File: rust/src/transformations/impute/mod.rs [hunk 2]
```diff
@@ -202,48 +214,59 @@ impl<T: CheckAtom + InherentNull + Clone> DropNullDomain for AtomDomain<T> {
...
 ///
 ///
-/// | `DA`                                | `DA::Imputed` |
-/// | ----------------------------------- | ------------- |
-/// | `OptionDomain<AtomDomain<TA>>`      | `TA`          |
-/// | `AtomDomain<TA>`                    | `TA`          |
+/// | input_domain                     |
+/// | -------------------------------- |
+/// | `option_domain(atom_domain(TA))` |
+/// | `atom_domain(TA)`                |
```
Comment (andrewvyrros) [inline rust/src/transformations/impute/mod.rs:225]: I think you need `vector_domain()` on these?

File: rust/src/transformations/impute/mod.rs (hunk 1 above)
Comment (Shoeboxam) [reply rust/src/transformations/impute/mod.rs:23]: There's only one valid input domain in this case, so the table feels kind of silly.

**Review by andrewvyrros — APPROVED** [review]

**Pull request merged**